src/main/scala/com/ascariandrea/revolut/sdk/client/Client.scala

```diff
@@ -0,0 +1,58 @@
+package com.ascariandrea.revolut.sdk
+package client
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.Uri.Path
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes, Uri}
+import akka.http.scaladsl.{Http, model}
+import akka.stream.ActorMaterializer
+import io.circe.Decoder
+
+import scala.concurrent.Future
+import io.circe.parser._
+
+class Client(baseUrl: Uri) {
+
+  implicit val system = ActorSystem()
+
+  implicit val materializer = ActorMaterializer()
+
+  implicit val executionContext = system.dispatcher
+
+  def get[T: Decoder](path: Path): Future[Either[HttpResponse, Option[T]]] = {
+    makeRequest(path)
+      .map((either: Either[HttpResponse, Option[String]]) =>
+        either.right.map(opt => opt.map(decode[T]).flatMap(_.toOption)))
+  }
+
+  def getMany[T: Decoder](
+      path: Path): Future[Either[HttpResponse, Option[List[T]]]] = {
+    makeRequest(path)
+      .map((either: Either[HttpResponse, Option[String]]) =>
+        either.right.map[Option[List[T]]](opt =>
+          opt.map(jsValue => {
+            parse(jsValue) match {
```

Comment:
this can become something like (you can easily find the right imports)

``` scala
import io.circe.parser.decode
jsValue.decode[List[T]]
```
(you dont need neither `parse`, nor mapping on the list)